internal/querynodev2/segments/pool.go

```diff
@@ -74,15 +93,19 @@ func initSQPool() {
 
 func initDynamicPool() {
 	dynOnce.Do(func() {
+		size := hardware.GetCPUNum()
 		pool := conc.NewPool[any](
-			hardware.GetCPUNum(),
+			size,
 			conc.WithPreAlloc(false),
 			conc.WithDisablePurge(false),
-			conc.WithPreHandler(runtime.LockOSThread), // lock os thread for cgo thread disposal
+			conc.WithPreHandler(func() {
+				runtime.LockOSThread()
+				C.SetThreadName(cgoTagDynamic)
```

Comment:
Can we make this a mandatory param?

Reply:
WithPreHandler？Since its a general worker pool, it is not recommend to make CGO related params mandatory